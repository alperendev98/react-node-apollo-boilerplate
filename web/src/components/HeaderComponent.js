import React from 'react';
import Menu from './MenuComponent'
import { getBEMClasses } from '../helpers/cssClassesHelper';

import '../styles/base/base.css';
import '../styles/components/headerComponent.css';

import magnifierIcon from '../assets/images/magnifier-tool.svg';

const BaseClasses = getBEMClasses(['base']);
const HeaderClasses = getBEMClasses(['header']);

class HeaderComponent extends React.Component {
	render() {
		return (
			<div className={HeaderClasses('container')}>
				<div className={HeaderClasses('account')}>
					<img
						className={HeaderClasses('account-icon')}
						src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRJ3cF4MlqomtuIptSvbEn3QQyhlmdNEB-SAFDzXA7OcdfR2LziVw"
					/>
				</div>
				<div className = { HeaderClasses('menu') }>
					<Menu />				
				</div>
				<div className={HeaderClasses('filter')}>
					<input type="text" className={HeaderClasses('filter-box')} placeholder="Search" />
					<img className={HeaderClasses('filter-icon')} src={magnifierIcon} />
				</div>
			</div>
		);
	}
}

export default HeaderComponent;
