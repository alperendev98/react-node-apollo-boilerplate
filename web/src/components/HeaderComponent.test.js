import React from 'react';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import HeaderComponent from './HeaderComponent';

const mockStore = configureStore();

describe('HeaderComponent', () => {
	const initialState = {};

	let store, component, wrapper;

	beforeEach(() => {
		store = mockStore(initialState);
		component = mount(<HeaderComponent store={store} />, {});
		wrapper = shallow(<HeaderComponent />);
	});

	it('shallowly renders EquipmentComponent without crashing', () => {
		expect(wrapper.length).toEqual(1);
	});

	it('should render HeaderComponent with .header__container class', () => {
		expect(component.find('.header__container').length).toBe(1);
	});

	it('should match previos snapshot for HeaderComponent', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
