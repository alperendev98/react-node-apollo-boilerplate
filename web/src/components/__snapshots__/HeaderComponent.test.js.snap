// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should match previos snapshot for HeaderComponent 1`] = `
<div
  className="header__container"
>
  <div
    className="header__account"
  >
    <img
      className="header__account-icon"
      src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRJ3cF4MlqomtuIptSvbEn3QQyhlmdNEB-SAFDzXA7OcdfR2LziVw"
    />
  </div>
  <div
    className="header__menu"
  >
    <MenuComponent />
  </div>
  <div
    className="header__filter"
  >
    <input
      className="header__filter-box"
      placeholder="Search"
      type="text"
    />
    <img
      className="header__filter-icon"
      src="magnifier-tool.svg"
    />
  </div>
</div>
`;
